nasa/openmct
# [Tabs View] add ability to remove tabs from tabs view interface (#3147)

Fixes #3147 

* an "X" on each tab is visible in edit mode

Author Checklist:
  1. Changes address original issue? Yes
  2. Unit tests included and/or updated with changes? No
  3. Command line build passes? Yes
  4. Changes have been smoke-tested? Yes
  5. Testing instructions included? Yes

Test:
1. add one or more tabs to tabe view object and save
2. ensure there are no "X" icons on the tabs when not in edit mode
3. go into edit mode and ensure each tab has an "X"
4. Click the "X" to remove a tab and ensure a confirmation dialog appears.
5. Ensure the dialog Cancel button dismisses the dialog.
6. Ensure the dialog OK button removes the tab from the tabs view UI and from the object tree
7. Cancel tab removal by clicking the "X" in the tabs view toolbar and ensure the tab reappears in the tabs view UI and in the tabs view object tree
8. Remove a tab and save the tabs view this time. Ensure the tab is removed from the tabs view UI and the object is removed from the tabs view object tree.

![Screenshot from 2020-07-01 22-08-56](https://user-images.githubusercontent.com/65196434/86315566-a519ea80-bbe7-11ea-9cfd-40e70b8c7717.png)

![Screenshot from 2020-07-01 22-16-19](https://user-images.githubusercontent.com/65196434/86315845-9122b880-bbe8-11ea-93ef-b7dd487eaebf.png)
